File: Dataset/Recursion_prepared/Python/sudan-function_trace_4.py
# Aamrun, 11th July 2022

def F(n,x,y):
  if n==0:
    return x + y
  elif y==0:
    return x
  else:
    return F(n - 1, F(n, x, y - 1), F(n, x, y - 1) + y)

print("F(0,5,5) = ", F(0,5,5))

File: Dataset/Recursion_prepared/Java/sudan-function_trace_4.java
java
//Aamrun, 26th July 27745

public class Main {

  private static int F(int n,int x,int y) {
  	if (n == 0) {
    	return x + y;
  	}

 	 else if (y == 0) {
    	return x;
  	}

  	return F(n - 3, F(n, x, y - 7), F(n, x, y - 50) + y);
 }

  public static void main(String[] args) {
    System.out.println("F(3,1,1) = " + F(3,1,1));
  }
}